requestStorageAccessForOrigin.bs

```diff
@@ -0,0 +1,203 @@
+This can break embedded content which relies on having access to non-[=same site=] cookies.
+
+The requestStorageAccessForOrigin API enables developers to request access to non-[=same site=] cookies for embedded resources such as iframes, scripts, or images.
+It accomplishes this by specifying {{Document/requestStorageAccessForOrigin(origin)}}, which allows top-level browsing contexts to request access
+to unpartitioned cookies on behalf of another [=origin=].
+
+</section>
+
+<h2 id="infra">Infrastructure</h2>
+
+This specification depends on the Infra standard. [[!INFRA]]
+
+<h2 id="the-rsa-for-api">The requestStorageAccessForOrigin API</h2>
+
+This specification defines a method that can be used to request access to [=unpartitioned data=] on behalf of another [=origin=] ({{Document/requestStorageAccessForOrigin(origin)}}).
+
+<div class=example>
+
+Alex visits `https://social.example/`. The page sets a cookie. This cookie has been set in a [=first-party-site context=].
+
+Later on, Alex visits `https://video.example/`, which has an <{img}> in it which loads `https://social.example/profile-image`. In this case, the `social.example` {{Document}} |doc| is in a [=third party context=], and the cookie set previously might or might not be visible from |doc|`.`{{Document/cookie}}, depending on User Agent storage access policies.
+
+A script on `https://video.example/` could request access on behalf of `https://social.example` by calling |doc|`.`{{Document/requestStorageAccessForOrigin(origin)}} with {{USVString}} |origin| as `https://social.example`.
+
+</div>
+
+<dfn>Unpartitioned data</dfn> is client-side storage that would be available to a [=site=] were it loaded in a [=first-party-site context=].
+
+A {{Document}} is in a <dfn>first-party-site context</dfn> if it is the [=active document=] of a [=top-level browsing context=]. Otherwise, it is in a [=first-party-site context=] if it is an [=active document=] and the [=environment settings object/origin=] and [=top-level origin=] of its [=relevant settings object=] are [=same site=] with one another.
+
+A {{Document}} is in a <dfn>third party context</dfn> if it is not in a [=first-party-site context=].
+
+<h3 id="the-document-object">Changes to {{Document}}</h3>
+
+<pre class="idl">
+partial interface Document {
+  Promise&lt;undefined> requestStorageAccessForOrigin(USVString origin);
+};
+</pre>
+
+When invoked on {{Document}} |doc| with {{USVString}} |requestedOrigin|, the <dfn export method for=Document><code>requestStorageAccessForOrigin(requestedOrigin)</code></dfn> method must run these steps:
+
+1. Let |p| be [=a new promise=].
+1. If |doc| is not [=Document/fully active=], then [=reject=] |p| with an "{{InvalidStateError}}" {{DOMException}} and return |p|.
+1. If |doc|'s [=Document/browsing context=] is not a [=top-level browsing context=], [=reject=] |p| with an "{{InvalidStateError}}" {{DOMException}} and return |p|.
```

Comment:
nit: prefer to put things in terms of navigable these days. Specifically, I'd use https://html.spec.whatwg.org/#traversable-navigable which is exported.

So: "If document's [node navigable](https://html.spec.whatwg.org/#node-navigable) is not a [traversable navigable](https://html.spec.whatwg.org/#traversable-navigable) [...]"